package/samplers/moead/moead.py

```diff
@@ -0,0 +1,224 @@
+from __future__ import annotations
+
+from collections import defaultdict
+import hashlib
+from typing import Any
+from typing import Dict
+from typing import TYPE_CHECKING
+
+import optuna
+from optuna.distributions import BaseDistribution
+from optuna.samplers import BaseSampler
+from optuna.samplers import RandomSampler
+from optuna.samplers._lazy_random_state import LazyRandomState
+from optuna.samplers.nsgaii._crossovers._base import BaseCrossover
+from optuna.samplers.nsgaii._crossovers._uniform import UniformCrossover
+from optuna.search_space import IntersectionSearchSpace
+from optuna.trial import FrozenTrial
+
+from ._child_generation_strategy import MOEAdChildGenerationStrategy
+from ._elite_population_selection_strategy import MOEAdElitePopulationSelectionStrategy
+
+
+if TYPE_CHECKING:
+    from optuna.study import Study
+
+# Define key names of `Trial.system_attrs`.
+_GENERATION_KEY = "moead:generation"
+_POPULATION_CACHE_KEY_PREFIX = "moead:population"
+
+
+class MOEADSampler(BaseSampler):
+    def __init__(
+        self,
+        *,
+        population_size: int = 100,
+        n_neighbors: int | None = None,
+        scalar_aggregation_func: str = "tchebycheff",
```

Comment:
`scalar_aggregation_func` seems to be either `tchebycheff` or `weighted_sum`, we may be use `Literal` as a type hint.

```suggestion
        scalar_aggregation_func: Literal['weighted_sum', 'tchebycheff'] = "tchebycheff",
```